### ISSUE TITLE
WordPress 5.1:  Warning: array_map(): Expected parameter 2 to be an array, string given

### ISSUE THREAD

**Issue by connerbw** [johnbillion/extended-cpts#116]
Issue:

 + I can call `register_extended_taxonomy` with `dropdown`
 + I can call with `hierarchical` set to `true`

```php
register_extended_taxonomy(
	'chapter-type',
	'chapter',
	[
		'meta_box' => 'dropdown',
		'hierarchical' => true,
		// ... snip ...
```				

Since upgrading to WordPress 5.1.0, when I `$_POST`, I get a PHP error:

> ( ! ) Warning: array_map(): Expected parameter 2 to be an array, string given in wp-admin/includes/post.php on line 2005
 
 ```php
 /**
 * Sanitizes POST values from a checkbox taxonomy metabox.
 *
 * @since 5.1.0
 *
 * @param mixed $terms Raw term data from the 'tax_input' field.
 * @return array
 */
function taxonomy_meta_box_sanitize_cb_checkboxes( $taxonomy, $terms ) {
	return array_map( 'intval', $terms );
}
```

The POST looks like: `tax_input[chapter-type]: -1`
The value of `$terms` passed to `taxonomy_meta_box_sanitize_cb_checkboxes` is equal to `-1`

Commit that broke things:
 + https://github.com/WordPress/wordpress-develop/commit/4ec5d65dcc2e480506c22dc77e56af8a6334a33e

Change in 5.1.0 that weren't in 5.0.3
 + https://github.com/WordPress/wordpress-develop/blob/5.1.0/src/wp-includes/class-wp-taxonomy.php#L360
 + https://github.com/WordPress/wordpress-develop/blob/5.0.3/src/wp-includes/class-wp-taxonomy.php#L346

**Comment by connerbw** [comment]
I can fix the issue by setting:

`'meta_box_sanitize_cb' => 'taxonomy_meta_box_sanitize_cb_input',`

Maybe extended-cpts can just do this automatically when meta_box is dropdown?

More likely, the `taxonomy_meta_box_sanitize_cb_checkboxes` needs to be fixed.

The param says ` @param mixed $terms` ... but that's now how `array_map` works.

**Comment by connerbw** [comment]
WordPress Core Ticket: https://core.trac.wordpress.org/ticket/46338

**Comment by johnbillion** [comment]
This should be fixed by #131
